### ISSUE TITLE
Reporting Module: Create a table for the operations list on "Reporting" Page

### ISSUE THREAD

**Issue by mikevespi** [bcgov/cas-reporting#217]
#### Description:

Blocked by: #123 

As a part of #123 we have a page showing operators a list of their operations. We need to convert this list into a table similar to the one we present operators in Registration. In the [wireframe](https://www.figma.com/file/LsDmLDCdnJqI8UahyqvOD8/OBPS-App-Designs?type=design&node-id=7789%3A2031&mode=design&t=mcPbilWJY8esC5XQ-1) some of the columns of the table will require more backend prep. This card is focused on converting the relevant information that we are already being returned into a table, and the other columns have been split out into their own cards. 

The columns the table should start with are simply:
- BC GHG ID (I added thisbecause it is already being returned, and to maintain a consistent representation of operations across the site, this is what registration use as the identifier in their table)
- Operation (this is the operation name)
- Action (this will contain a dead link for now, there's a card to populate it with the various action)

The Registration page uses a component called `OperationsDataGrid`, and while it can't be directly used (the columns don't line up), the pattern can be copied. It is based on a component in the same directory (`/client/app/components`) called `DataGrid`. If this component is used / extended, it should be copied across to a common library.

#### Acceptance Criteria:

As a logged-in external operator,
When I navigate to the Reporting module,
I am presented with a table of all of my operations, their names, and BC GHG IDs

- [x] The list of operations on the Reporting page has been converted to a table on the page
- [x] The table of operations has the BC GHG ID, Operation (name), and Action columns
- [x] The table is populated with precisely the operations that the logged in external operator has access to, and the appropriate values are in their respective fields
- [x] The style of the table matches that of the one found in the Registration module
- [x] The table columns BC GHG ID and Operation are sortable

#### Development Checklist:

- [ ] A new data grid based component exists for the Reporting page (note while the DataGrid component may make sense to bring into a shared library, this Reporting specific operations data grid is indeed reporting specific)
- [ ] The list of operations on the Reporting page has been converted to the new data grid component, with the fields indicated in AC
- [ ] Any new components have been appropriately tested
- [ ] Any components from the Registration code base that were reused have been moved to a shared library in BCIERS
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by pbastia** [comment]
We might want to look into making the registration table component an Nx component

**Comment by rdromey** [comment]
@Mike to confirm with business area and Moose and will refine with the information.

**Comment by mikevespi** [comment]
A word of caution to whoever picks this up: in the return from `registration/operations` you'll see a status; don't be deceived and think you've got a freebie here. It's the status of registration, not reporting.

**Comment by dleard** [comment]
@pbastia @BallardRobinett @patriciarussellCAS 
I think we need a follow up card to pull our snapshot of operation/facility data on clicking "start" in this and save it to report-specific tables. I think this potentially blocks #196 (and in turn #195) (or we build out those forms/pages with no data coming from the backend)

**Comment by pbastia** [comment]
@dleard @BallardRobinett @patriciarussellCAS  I don't think we have a "I want to create a report" user story yet, I'm guessing the freezing of that data in a report specific structure will happen there?

**Comment by patriciarussellCAS** [comment]
Hey @BallardRobinett and @pbastia - are all the AC checks complete? And the DODs? Just want to confirm before closing this. 
Thanks!
